xwiki-rendering-macros/xwiki-rendering-macro-toc/src/main/java/org/xwiki/rendering/internal/macro/toc/TocMacro.java

```diff
@@ -123,7 +124,13 @@ public boolean supportsInlineMode()
 
         TreeParametersBuilder builder = new TreeParametersBuilder();
         TreeParameters treeParameters = builder.build(rootBlock, parameters, context);
-        return this.tocTreeBuilder.build(treeParameters);
+        List<Block> parametersList = this.tocTreeBuilder.build(treeParameters);
+        if (parameters.getBox()) {
+            MacroBlock mb = new MacroBlock("toc", parametersList.get(0).getRoot().getParameters(), false);
```

Comment:
I don't think this does what you expect. You are return a new toc macro here, not a box which seems to be the intent of this `if`.

Reply:
@tmortagne I'm not sure how I would go about creating a box, do I create a BoxMacro object?

Reply:
You can look at BoxMacro code to see how it's butting a box around the content.

Reply:
Yes but don't copy that. You need to reuse the Box Macro instead of copying it to avoid duplication and also to benefit from its future improvements.